src/main/java/com/nttdata/estore/controllers/OrderController.java

```diff
@@ -24,10 +27,24 @@
 
     @GetMapping(path = "/orders/{id}")
     public ResponseEntity getOrder(@PathVariable("id") int id) {
-        Order order = orderService.getById(id);
+        Order order = orderService.getOrder(id);
         if (null == order) {
             return new ResponseEntity("No Order found for ID " + id, HttpStatus.NOT_FOUND);
         }
         return new ResponseEntity(order, HttpStatus.OK);
     }
-}
+
+    @PutMapping(path = "/orders/{id}")
+    public ResponseEntity updateOrder(@PathVariable int id) {
+        // TODO check method
+        Order oldOrder = orderService.getOrder(id);
+        if(oldOrder.getOrderStatus().name().equals("WAITING") || oldOrder.getOrderStatus().name().equals("OPEN") ) {
+            oldOrder.setOrderStatus(OrderStatus.CANCELED);
+        }
+            orderService.saveOrUpdateOrder(oldOrder);
+        if (null == oldOrder) {
```

Comment:
verificarea aceasta ar fi bine sa fie inainte de if-ul anterior (cel din linia 41). Altfel primesti NullPointerException in linia 41 daca oldOrder = null :)